community/kernel/src/main/java/org/neo4j/kernel/api/security/AuthToken.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api.security;
+
+import java.util.Map;
+
+import org.neo4j.kernel.api.security.exception.InvalidAuthTokenException;
+
+public interface AuthToken
+{
+    String SCHEME_KEY = "scheme";
+    String PRINCIPAL = "principal";
+    String CREDENTIALS = "credentials";
+    String NEW_CREDENTIALS = "new_credentials";
+
+    static String safeCast( String key, Map<String,Object> authToken ) throws InvalidAuthTokenException
```

Comment:
The map comes from the client and you could theoretically use any neo4j type as value. It is our built-in basic username-password scheme that requires strings, but we want to avoid unnecessary restrictions since we want to allow a pluggable authentication realm to handle a token from a client application with an externally defined contract.